barretenberg_wasm/src/blake2s.rs

```diff
@@ -1,25 +1,26 @@
-use super::Barretenberg;
+use super::{Barretenberg, FIELD_BYTES};
 use common::acvm::FieldElement;
 use wasmer::Value;
 
 impl Barretenberg {
     // TODO : Replace this call with a blake2s call and a field element reduce
     /// Hashes to a bn254 scalar field element using blake2s
     pub fn hash_to_field(&mut self, input: &[u8]) -> FieldElement {
-        let input_ptr = self.allocate(input); // 0..32
-
-        let result_ptr = Value::I32(0);
-
-        // Not sure why this is needed to send to WASM
-        // It seems to be sent twice?
-        let data_len = Value::I32(input.len() as i32);
-
-        self.call_multiple("blake2s_to_field", vec![&input_ptr, &data_len, &result_ptr]);
+        let input_ptr = self.allocate(input);
+        let result_ptr: usize = 0;
+
+        self.call_multiple(
+            "blake2s_to_field",
+            vec![
+                &input_ptr,
+                &Value::I32(input.len() as i32),
+                &Value::I32(result_ptr as i32),
+            ],
+        );
 
         self.free(input_ptr);
 
-        // result_ptr is 0 and the output is 32 bytes long
-        let result_bytes = self.slice_memory(0, 32);
+        let result_bytes = self.slice_memory(result_ptr, result_ptr + FIELD_BYTES);
```

Comment:
Just a thought, I wonder if we should change the API slightly such that we don't need to do the arithmetic here, it would then be:

```rust
self.slice_memory(result_ptr, FIELD_BYTES);
```

i.e. instead of giving the absolute address for start and end, you just give the address and then an offset.